mozetl/bhr_collection/bhr_collection.py

```diff
@@ -0,0 +1,1359 @@
+        )
+        if data is None:
+            print("No data")
+            continue
+        transformed, usage_hours = transform_pings(sc, data, final_config)
+        profile_processor = ProfileProcessor(final_config)
+        profile_processor.ingest(transformed, usage_hours)
+        profile = profile_processor.process_into_profile()
+        filepath = "%s_%s" % (final_config["hang_profile_out_filename"], date_str)
+        print("writing file %s" % filepath)
+        write_file(filepath, profile, final_config)
+        filepath = "%s_current" % final_config["hang_profile_out_filename"]
+        print("writing file %s" % filepath)
+        write_file(filepath, profile, final_config)
+        gc.collect()
+        print_progress(job_start, iterations, x, iteration_start, date_str)
+
+
+def etl_job_incremental_finalize(_, __, config=None):
+    final_config = {}
+    final_config.update(default_config)
+
+    if config is not None:
+        final_config.update(config)
+
+    if final_config["hang_profile_out_filename"] is None:
+        final_config["hang_profile_out_filename"] = final_config[
+            "hang_profile_in_filename"
+        ]
+
+    profile_processor = ProfileProcessor(final_config)
+    iterations = (final_config["end_date"] - final_config["start_date"]).days + 1
+    job_start = time.time()
+    current_date = None
+    for x in range(iterations):
+        iteration_start = time.time()
+        current_date = final_config["start_date"] + timedelta(days=x)
+        date_str = current_date.strftime("%Y%m%d")
+        profile = read_file(
+            "%s_incremental_%s" % (final_config["hang_profile_in_filename"], date_str),
+            final_config,
+        )
+        profile_processor.ingest_processed_profile(profile)
+        gc.collect()
+        print_progress(job_start, iterations, x, iteration_start, date_str)
+
+    if final_config["split_threads_in_out_file"]:
+        profile = profile_processor.process_into_split_profile()
+        for files in profile["file_data"]:
+            for name, data in files:
+                write_file(
+                    final_config["hang_profile_out_filename"] + "_" + name,
+                    data,
+                    final_config,
+                )
+        write_file(
+            final_config["hang_profile_out_filename"],
+            profile["main_payload"],
+            final_config,
+        )
+    else:
+        profile = profile_processor.process_into_profile()
+        write_file(final_config["hang_profile_out_filename"], profile, final_config)
+
+
+@click.command()
+@click.option(
+    "--sample-size",
+    type=float,
+    default=0.5,
+    help="Proportion of pings to use (1.0 is 100%)",
+)
+def start_job(sample_size):
+    print(f"Using sample size {sample_size}")
+    etl_job_daily(
+        sc,
+        spark,
+        {
+            "start_date": datetime.today() - timedelta(days=3),
```

Comment:
This command should take the start_date parameter as an argument so the Airflow dag can take over what date ranges are run.